tests/framework/extensions/nodes/ec2/ec2.go

```diff
@@ -19,111 +19,232 @@ func CreateNodes(client *rancher.Client, numOfInstances int) ([]*nodes.Node, err
 	if err != nil {
 		return nil, err
 	}
-
-	sshName := getSSHKeyName(ec2Client.Config.AWSSSHKeyName)
-
-	runInstancesInput := &ec2.RunInstancesInput{
-		ImageId:      aws.String(ec2Client.Config.AWSAMI),
-		InstanceType: aws.String(ec2Client.Config.InstanceType),
-		MinCount:     aws.Int64(int64(numOfInstances)),
-		MaxCount:     aws.Int64(int64(numOfInstances)),
-		KeyName:      aws.String(sshName),
-		BlockDeviceMappings: []*ec2.BlockDeviceMapping{
-			{
-				DeviceName: aws.String("/dev/sda1"),
-				Ebs: &ec2.EbsBlockDevice{
-					VolumeSize: aws.Int64(int64(ec2Client.Config.VolumeSize)),
+	var ec2Nodes []*nodes.Node
+	for _, config := range ec2Client.ClientConfig.AWSEC2Config {
+		if !config.IsWindows {
```

Comment:
Is this test for windows nodes in ec2? If so, why are we checking that the config is not windows?

Reply:
I'm confused about this too, if this is to create linux node, why do we need this if statement?

Reply:
I have restructured the code so that this `if` statement is gone, and the check for `isWindows` makes more sense now. I was also able to strip out a TON of no longer needed code as a result of my refactor.